compiler/src/dmd/dsymbolsem.d

```diff
@@ -6292,7 +6292,17 @@ void templateInstanceSemantic(TemplateInstance tempinst, Scope* sc, ArgumentList
     sc2.parent = tempinst;
     sc2.tinst = tempinst;
     sc2.minst = tempinst.minst;
-    sc2.stc &= ~STC.deprecated_;
+    if (sc.isDeprecated() && tempinst.name == Id._d_arrayappendcTXImpl)
```

Comment:
why single case arrayappend here? Shouldn't we do this for all hooks?

Reply:
Right. In this case, should I add functions that test all of them to `test23965.d`?

Reply:
I don't think that's necessary.